src/cmd/troubleshoot/image_helper.go

```diff
@@ -0,0 +1,138 @@
+package troubleshoot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	removeSchemaRegex      = regexp.MustCompile("^.*//(.*)$")
+	removeApiEndpointRegex = regexp.MustCompile("^(.*)/[^/]*$")
+	registryRegex          = regexp.MustCompile(`^(.*)/linux.*$`)
+	imageRegex             = regexp.MustCompile(`^.*/(linux.*)$`)
+)
+
+type imageInfo struct {
+	registry string
+	image    string
+	version  string
+}
+
+// Split image path in into its components.
+// Some image path examples that work with this function:
+//
+//	aaa00000.dynatrace.com/linux/oneagent
+//	aaa00000.dynatrace.com/linux/activegate:1.123
+func splitImageName(imageName string) (imageInfo, error) {
```

Comment:
Parsing image could be moved to shared package so it can be reused.
We already have some kind of image parsing implemented here: https://github.com/Dynatrace/dynatrace-operator/blob/master/src/installer/image/source.go#L30

Reply:
Wil note it down with all other non PR related findings.